synthesis/population/spatial/primary/locations.py

```diff
@@ -106,13 +119,18 @@ def execute(context):
     df_work_candidates = pd.merge(df_work_candidates, df_locations, how = "left", on = "location_id")
     df_work_candidates = gpd.GeoDataFrame(df_work_candidates)
 
-    df_locations = context.stage("synthesis.locations.education")[["location_id", "geometry"]]
+    df_locations = context.stage("synthesis.locations.education")[["TYPEQU", "location_id", "geometry"]]
     df_education_candidates = data["education_candidates"]
     df_education_candidates = pd.merge(df_education_candidates, df_locations, how = "left", on = "location_id")
     df_education_candidates = gpd.GeoDataFrame(df_education_candidates)
 
     # Assign destinations
     df_work = process(context, "work", df_work, df_work_candidates)
-    df_education = process(context, "education", df_education, df_education_candidates)
-
+    if context.config("education_location_source") == 'bpe':
+        df_education = process(context, "education", df_education, df_education_candidates)
+    else :
+        education = []
+        for prefix, education_type in EDUCATION_MAPPING.items():
+            education.append(process(context, "education_" + prefix,df_education[df_education["age"].between(education_type["min_age"],education_type["max_age"])],df_education_candidates[df_education_candidates["TYPEQU"].str.startswith(education_type["type_edu"])]))
+        df_education = pd.concat(education).sort_index()
```

Comment:
Same comment as above, I think this could be simplified a bit. Especially we could assign an age_range attribute to the persons at some point so everything just becomes a matching by that attribute instead of doing an `between` filter each time.